src/Http/Client.php

```diff
@@ -84,21 +85,27 @@ public function get($path, $query = [])
     }
 
     /**
-     * @param string $path
-     * @param null   $body
-     * @param array  $query
+     * @param mixed|null $body
      *
      * @return mixed
      *
-     * @throws ClientExceptionInterface
      * @throws ApiException
+     * @throws ClientExceptionInterface
+     * @throws CommunicationException
+     * @throws FailedJsonEncodingException
      */
-    public function post($path, $body = null, $query = [])
+    public function post(string $path, $body = null, array $query = [])
     {
+        $content = json_encode($body);
```

Comment:
Better late than never :) I'd suggest instead to leverage the `JSON_THROW_ON_ERROR`

Reply:
Good point. I guess a PR is more than welcome 😄

Reply:
@norkunas thanks for the catch!
I agreed with @codedge if you are comfortable with the idea of making a pr it would indeed be very welcome.

Reply:
in progress

Reply:
#185 :)